### PR TITLE
feat: add analytics

### PR DESCRIPTION
**Pull request by asanehisa**
Tested via local dev + local prod
 Tested in deploy [here](https://www.yext.com/s/4079215/yextsites/131259/branch/93817/deploys/recent)
 
 with `enabledDebugging={true}`, here are some of the analytics printed in the console.log: 
 Tracked Pages event: PAGE_VIEW
 Tracked Pages event: link
 Tracked Pages event: cta
 Tracked Pages event: header_cta
 Tracked Pages event:  footer_link
Tracked Pages event: header_link

### CODE REVIEW

**Comment by jwartofsky-yext** [comment]
> Tested via local dev + local prod Tested in deploy [here](https://www.yext.com/s/4079215/yextsites/131259/branch/93817/deploys/recent)

The code looks correct, but when I click the link I get taken to a Terms and Conditions page that doesn't let me move forward. If I check the box, scroll to the bottom, and confirm it just refreshes the page.

**Comment by asanehisa** [comment]
> > Tested via local dev + local prod Tested in deploy [here](https://www.yext.com/s/4079215/yextsites/131259/branch/93817/deploys/recent)
> 
> The code looks correct, but when I click the link I get taken to a Terms and Conditions page that doesn't let me move forward. If I check the box, scroll to the bottom, and confirm it just refreshes the page.

Oh very odd. That doesn't happen on my end? I don't think I had it request user optIn either.
Screenshot 2024-01-18 at 2 20 59 PM

**Review by jwartofsky-yext — APPROVED** [review]
lgtm

File: src/components/PageLayout.tsx [hunk 1]
```diff
@@ -6,10 +6,20 @@ import { TemplateProps } from "@yext/pages";
 export interface PageLayoutProps {
   children?: React.ReactNode;
   _site?: any;
-  templateData: TemplateProps;
+  templateData?: TemplateProps;
 }
 
 const PageLayout = ({ children, _site, templateData }: PageLayoutProps) => {
+  if (!templateData) {
```
Comment (mkilpatrick) [inline src/components/PageLayout.tsx]: When is there no templateData?

Comment (asanehisa) [reply src/components/PageLayout.tsx]: Undid it, I accidentally thought static pages don't have templateData, but they do.

File: src/components/Cta.tsx [hunk 2]
```diff
@@ -9,7 +10,7 @@ export interface CtaProps {
 const Cta = ({ buttonText, url, style }: CtaProps) => {
   return (
     <>
-      <a
+      <Link
```
Comment (mkilpatrick) [inline src/components/Cta.tsx:13]: This should set the cta prop on Link considering this is a cta component

**Review by mkilpatrick — COMMENTED** [review]
Can you test this out and put some examples of what the analytics events are in the PR description?

**Review by mkilpatrick — APPROVED** [review]

**Pull request merged**